### DISCUSSION TITLE
What is the use of supabase?

### DISCUSSION THREAD

**Discussion by satyap54** [deleteman/the-hive#9] [Q&A] [answered]
Why do we upsert data everytime into supabase? All the edge-functions interact with github client first. My point is why do we store the data that we fetch from github into our db?

**Comment by deleteman** [comment, marked as answer]
Hey @satyap54 , multiple reasons actually:

1. Queries to Github api are slower and quota capped, meaning that we might be rendered unable to query the api if we overdo it during a month.
2. Queries to an external API like Github's are slower than to a database, even an external one like Supabase.
3. Some of the datasets queried are kept as historic data for the repo to showcase the evolution around certain aspects. And the github api only returns the actual data. So the historic record needs to be kept somewhere else.
4. If tomorrow we want to add another git provider, like Gitlabs, then we already have the storage layer and data models already defined. 

I hope that clears it up!

**Reply by satyap54** [reply]
So, when are the supabase functions triggered? My assumption is that we trigger them periodically to update the db.

**Reply by deleteman** [reply]
Correct, the idea is to trigger the main function ("gather-project-metrics") once a day, that function will invoke the rest.

**Reply by satyap54** [reply]
Should we write a Cron to do that? Would that be a good idea?

**Reply by deleteman** [reply]
Eventually we'll find a way to orchestrate that. Right now a cron wouldn't do any good because where would we put it? We don't have any servers.
I know supabase is working on a workflow service, so maybe by the time we need to orchestrate this, they already have it. Right now i wouldn't really spend time on this.

**Reply by satyap54** [reply]
Yes, makes sense

**Reply by JorchRL** [reply]
If at all relevant, I think it could be done with Supabase itself. As it supports the pg_cron postgres extension. Idk, maybe as a temporary solution. https://supabase.com/blog/2021/03/05/postgres-as-a-cron-server
